app/js/results/smaaResultsService.js

```diff
@@ -339,11 +340,29 @@ define([
       });
     }
 
+    function hasNoStochasticMeasurements(aggregateState) {
+      return !_.some(aggregateState.problem.performanceTable, function(tableEntry) {
+        return tableEntry.performance.distribution && tableEntry.performance.distribution.type !== 'exact';
+      });
+    }
+
+    function hasNoStochasticWeights(aggregateState) {
+      return aggregateState.prefs && aggregateState.prefs.length !== 0 && areAllPreferencesExact(aggregateState);
+    }
+
+    function areAllPreferencesExact(aggregateState) {
+      return !_.some(aggregateState.prefs, function(pref) {
+        return pref.type !== 'exact swing';
```

Comment:
this is correct for now but harder to maintain than a precise list of the preference types which are/are not stochastic.